test/onnx/test_pytorch_onnx_no_runtime.py

```diff
@@ -762,6 +762,84 @@ def check_proto():
 
         self.assertRaises(RuntimeError, check_proto)
 
+    def test_is_fp_for_C_TypeList(self):
+        def static_squeeze_11(g, self, dim=None):
+            """Simplified version of squeeze-11 without dynamic dims support
+
+            It does not create a `onnx::If` node when `self.shape[i]==1`.
+            That trick facilitates reproducing the bug on this simplified model
+            by not generating squeeze with unknown rank (caused by the `onnx::If`)
+            """
+
+            if dim is None:
+                return g.op("Squeeze", self)
+            if not symbolic_helper._is_constant(dim):
+                return symbolic_helper._squeeze_helper(g, self, [dim])
+            dim = symbolic_helper._get_const(dim, "i", "dim")
+            input_rank = symbolic_helper._get_tensor_rank(self)
+            adjusted_dim = dim
+            if input_rank is not None and dim < 0:
+                adjusted_dim += input_rank
+            dim_size = symbolic_helper._get_tensor_dim_size(self, adjusted_dim)
+            dim = adjusted_dim
+            if dim_size is not None and dim_size > 1:
+                return self
+            return symbolic_helper._squeeze_helper(g, self, [dim])
+
+        @torch.jit.script_if_tracing
+        def scriptable_function(maps, rois):
+            """Scriptable function meant to allow a `torch._C.TypeList.ofTensors()` input"""
+
+            size = (rois[:, 2] - rois[:, 0]).clamp(min=1)
+            _, num_keypoints = maps.shape[:2]
+            keypoints_idx = torch.arange(num_keypoints)
+            outsize = (int(size[0]), int(size[0]))
+
+            # Although F.interpolate results in a `roi_map` tensor with rank 4,
+            # the subsequent `roi_map.squeeze(0)` will result in its rank to be unknown.
+            # Therefore, any subsequent layer using `roi_map`'s shape causes export failure
+            roi_map = F.interpolate(
+                maps[[0]], size=outsize, mode="bicubic", align_corners=False
+            )
+
+            # Using a custom `squeeze-11` symbolic to bypass the unsupported indexing
+            # on `roi_map` tensor (due to unknown rank). Needed to reach the fixed
+            # `symbolic_helper._is_fp()` helper through `remainder-11`
+            roi_map = roi_map.squeeze(0)
```

Comment:
I wonder if it is ok to remove this line, such that we won't need the workaround symbolic in this test case.

Reply:
The actual fix for the model I was working on was replacing `squeeze` by `reshape`, but that doesn't change the limitation this PR addresses. It is valid to have either a `_C.TensorType` or `_C.ListType`, right?

Reply:
In case your question was regarding to the original bug, the final fix was actually replacing `squeeze(0)` by `reshape(tensor.shape[1:])`. In that case, although this is a valid workaround, it doesn't change the fact that a tenso can be either a `_C.Value` or `_C.ListType`, right? To have the model working I don't need this fix, but this PR makes exporter more robust